backend/common/repo.py

```diff
@@ -339,6 +369,7 @@ def _get_release_index_from_http(self) -> typing.Dict[str, "DpkgRepo.ReleaseEntr
             self._flat_checked = 1
 
             if not self.is_flat() and self.gpg_verify and not self._has_valid_gpg_signature(resp.url, resp):
+                logging.error("Repo has no valid GPG signature. Raising GeneralRepoException.")
                 raise GeneralRepoException("GPG verfication failed: {}".format(resp.url))
```

Comment:
Would it make sense to also include the actual problem in the exception that is raised? So instead of only saying "GPG verification failed", we could also say why it failed.

Reply:
Exceptions were already logged. But I've just added logging for the gpg return code.

Reply:
Yes, they were logged but did not offer help to what is broken. You could not look at the log and know why "GPG verification failed". I wonder why we don't pass that information to the exception, wouldn't it show up with the current logging we already do that way?